Add LockPremiumPromptsActivity for premium prompt management

rwhite27 opened the pull request:
Introduce LockPremiumPromptsActivity to handle the locking and unlocking of premium prompts based on SKU validation. This activity integrates with the existing workflow to ensure proper management of premium content.

rwhite27 committed:
feat: add LockPremiumPromptsActivity for managing premium prompt locking

rwhite27 committed:
fix: remove unused import for UserConfig in LockPremiumPromptsActivity

rwhite27 committed:
feat: add LockPremiumPromptsActivity to KanvasWorkflowSynActionCommand

rwhite27 committed:
feat: add message locking check for viewing one message in getAll method

kaioken commented on src/Domains/Connectors/PromptMine/Workflows/Activities/LockPremiumPromptsActivity.php:
```diff
@@ -0,0 +1,76 @@
+    #[Override]
+    public function execute(Model $entity, AppInterface $app, array $params): array
+    {
+        $messageData = ! is_array($entity->message) ? json_decode($entity->message, true) : $entity->message;
```
@rwhite27 after migraiton no need for this

rwhite27 committed:
feat: Update EC2 deployment actions to use latest SCP action version

fix: Refactor user deactivation logic to use DeactivateUserAction

refactor: Update order processing to use AppEnums for cart session management

feat: Introduce SendRotationEmailsAction for handling lead rotation email notifications

chore: Upgrade prism package to version 0.65.1

chore: Update phpstan to version 2.1.13

fix: Improve ID verification report handling and name extraction logic

feat: Add PremiumPromptFlagActivity for handling premium prompt requests

refactor: Enhance PromptImageFilterActivity to set entity privacy

chore: Clean up unused CreateLeadsProductFromReceiverJob

fix: Adjust ProductTypeService to handle required attributes correctly

feat: Add setPremium and setNotPremium methods to Message model

refactor: Update DynamicRuleWorkflow to use report for exception handling

rwhite27 committed:
refactor: Remove LockPremiumPromptsActivity class and update LinkMessageToOrderActivity for improved message handling

rwhite27 committed:
refactor: Clean up imports and remove unused code in LinkMessageToOrderActivity

rwhite27 committed:
refactor: Remove redundant message retrieval logic in getAll method of MessageBuilder
